packages/fetchai/skills/carpark_client/handlers.py

```diff
@@ -211,35 +210,40 @@ def _handle_match_accept(self, msg: FIPAMessage, dialogue: Dialogue) -> None:
         :param dialogue: the dialogue object
         :return: None
         """
-        logger.info(
-            "[{}]: received MATCH_ACCEPT_W_INFORM from sender={}".format(
-                self.context.agent_name, msg.counterparty[-5:]
+        strategy = cast(Strategy, self.context.strategy)
+        if strategy.is_ledger_tx:
+            logger.info(
+                "[{}]: received MATCH_ACCEPT_W_INFORM from sender={}".format(
+                    self.context.agent_name, msg.counterparty[-5:]
+                )
             )
-        )
-        info = msg.info
-        address = cast(str, info.get("address"))
-        proposal = cast(Description, dialogue.proposal)
-        tx_msg = TransactionMessage(
-            performative=TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT,
-            skill_callback_ids=["carpark_client"],
-            tx_id="transaction0",
-            tx_sender_addr=self.context.agent_addresses["fetchai"],
-            tx_counterparty_addr=address,
-            tx_amount_by_currency_id={
-                proposal.values["currency_id"]: -proposal.values["price"]
-            },
-            tx_sender_fee=0,
-            tx_counterparty_fee=0,
-            tx_quantities_by_good_id={},
-            ledger_id=proposal.values["ledger_id"],
-            info={"dialogue_label": dialogue.dialogue_label.json},
-        )
-        self.context.decision_maker_message_queue.put_nowait(tx_msg)
-        logger.info(
-            "[{}]: proposing the transaction to the decision maker. Waiting for confirmation ...".format(
-                self.context.agent_name
+            info = msg.info
+            address = cast(str, info.get("address"))
+            proposal = cast(Description, dialogue.proposal)
+            tx_msg = TransactionMessage(
+                performative=TransactionMessage.Performative.PROPOSE_FOR_SETTLEMENT,
+                skill_callback_ids=["carpark_client"],
+                tx_id="transaction0",
+                tx_sender_addr=self.context.agent_addresses[
+                    proposal.values["ledger_id"]
+                ],
+                tx_counterparty_addr=address,
+                tx_amount_by_currency_id={
+                    proposal.values["currency_id"]: -proposal.values["price"]
+                },
+                tx_sender_fee=0,
```

Comment:
Check with the weather skills what we did there, but I don't think these should be zero

Reply:
It's the same with the weather client.